web/src/transmission.js

```diff
@@ -375,7 +375,10 @@ export class Transmission extends EventTarget {
       case Prefs.RefreshRate: {
         clearInterval(this.refreshTorrentsInterval);
         const callback = this.refreshTorrents.bind(this);
-        const msec = Math.max(2, this.prefs.refresh_rate_sec) * 1000;
+        const msec =
+          Number.parseFloat(this.prefs.refresh_rate_sec) > 0
```

Comment:
`Number.parseFloat()` has no effect here because `this.prefs.refresh_rate_sec` is either already an integer, or some value that cannot be recognised by `parseFloat()`:

This is where it gets converted to an integer:

https://github.com/transmission/transmission/blob/3e9f5f614a875aae987235d3836d0ca555a80d80/web/src/prefs.js#L80-L82

So

1. Remove `parseFloat()` here, and
2. Optionally, if you think it will be useful, change `parseInt()` to `parseFloat()` in `prefs.js`.

Reply:
I'm observing half a second refresh with 0.5 as cookie value just fine, I think that regex is not expecting a decimal in floating point number to begin with so it falls through. It accepts the integer number but then the operation below looks useless if it won't see a value other than an integer number. I'm nuking that code so tell me if there's something with it that I missed.

The Number.parseFloat() I'm using right now is supposed to safeguard against string type so it returns NaN, eventually returns false in comparison statement.

Reply:
I think the code to read the pref values from cookies is problematic. It creates a need to wrestle over error checking like what we are doing now.

I've created #7001 that addresses this as an alternative. See what you think @Rukario.

Reply:
It's not alternative but continuation with further improvements. Thank you for maintaining the principle of this PR and pick up all the good changes we find so far.